libs/platform/src/lib/components/table/components/table-view-settings-dialog/table-view-settings-dialog.component.ts

```diff
@@ -0,0 +1,65 @@
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    Component,
+    ContentChildren,
+    forwardRef,
+    Input,
+    QueryList,
+    TemplateRef,
+    ViewChild
+} from '@angular/core';
+
+import { TableComponent } from '../../table.component';
+import { TableViewSettingsFilterComponent } from '../table-view-settings-filter/table-view-settings-filter.component';
+
+/**
+ * View settings dialog component.
+ *
+ * Used to link view settings filters options to the grid table.
+ *
+ * ```html
+ * <fdp-table #myTable></fdp-table>
+ * ...
+ * <fdp-table-view-settings-dialog [table]="myTable">
+ *     <fdp-table-view-settings-filter
+ *         column="status"
+ *         label="Status"
+ *         type="single-select"
+ *         values="[{value: 'filterValue', label: 'Filter label'}]">
+ *     </fdp-table-view-settings-filter>
+ * </fdp-table-view-settings-dialog>
+ * ```
+ * */
+@Component({
+    selector: 'fdp-table-view-settings-dialog',
+    template: '',
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TableViewSettingsDialogComponent implements AfterViewInit {
+    /** Reference to table component. */
+    @Input()
+    table: TableComponent;
+
+    /** @hidden */
+    @ContentChildren(forwardRef(() => TableViewSettingsFilterComponent))
+    filters: QueryList<TableViewSettingsFilterComponent>;
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this._addFiltersToTable();
+    }
+
+    /** @hidden */
+    _addFiltersToTable(): void {
+        if (!this.table) {
+            return;
+        }
+
+        this.filters.changes.subscribe(() => {
+            this.table._setViewSettingsFilters(this.filters.toArray());
+        });
+
+        this.table._setViewSettingsFilters(this.filters.toArray());
+    }
```

Comment:
What about using startWith, instead of second call of `_setViewSettingsFilters`? Also you can get rid of brackets inside subscribe

Reply:
Added startWith